pkg/platform/runtime/setup/setup.go

```diff
@@ -420,20 +412,17 @@ func (s *Setup) installFromBuildLog(buildResult *model.BuildResult, artifacts ar
 
 // setupArtifact sets up an individual artifact
 // The artifact is downloaded, unpacked and then processed by the artifact setup implementation
-func (s *Setup) setupArtifact(buildEngine model.BuildEngine, a artifact.ArtifactID, unsignedURI string) error {
-	as, err := s.selectArtifactSetupImplementation(buildEngine, a)
-	if err != nil {
-		return errs.Wrap(err, "Failed to select artifact setup implementation")
-	}
-
+func (s *Setup) setupArtifact(a artifact.ArtifactID, unsignedURI string, as ArtifactSetuper) error {
 	targetDir := filepath.Join(s.store.InstallPath(), constants.LocalRuntimeTempDirectory)
 	if err := fileutils.MkdirUnlessExists(targetDir); err != nil {
 		return errs.Wrap(err, "Could not create temp runtime dir")
 	}
 
 	unarchiver := as.Unarchiver()
 	var archivePath string
-	if cachedPath, found := s.artifactCache.Get(a); found {
+	if s.target.InstallFromDir() != nil {
+		archivePath = unsignedURI
```

Comment:
So `unsignedURI` can be a path or an URI? That feels a little dirty and error prone. Please either refactor the function or use the arguments as intended. This might pass the type checking but it's not sane.

Reply:
Are you okay with `file://` in order to make it a URI instead of a path?

Reply:
I think that still has the same problem where the treatment of this variable is based on special hidden logic rather than logic that's conveyed through the function arguments themselves.

Can we create a new `setupArtifactFromURI()` function or something, which then calls `setupArtifact()` once it has downloaded the artifact and has a usable path?

Basically I don't want this argument to hold a value that depending on the way it is invoked is used in two completely distinctly different ways. That's hamfisting behaviour into the pre-existing function arguments.